server/src/main/java/org/elasticsearch/plugins/internal/DocumentParsingSupplier.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.plugins.internal;
+
+/**
+ * An interface to provide instances of document parsing observer and reporter
+ */
+public interface DocumentParsingSupplier {
```

Comment:
nit: supplier is normally for a singular thing being supplied. We've typically used "provider" (SPI terminology) for something like this which provides many things.